File: elixir/bob/bob.exs
defmodule Teenager do
  @moduledoc """
  Bob is a lackadaisical teenager

  Bob answers 'Sure.' if you ask him a question.

  He answers 'Woah, chill out!' if you yell at him.

  He says 'Fine. Be that way!' if you address him without actually saying anything.

  He answers 'Whatever.' to anything else.
  """

  def hey(string) do
    cond do
      silence?(string)  -> "Fine. Be that way!"
      yelling?(string)  -> "Woah, chill out!"
      question?(string) -> "Sure."
      true              -> "Whatever."
    end
  end

  defp yelling?(string) do
    String.upcase(string) == string && String.match?(string, ~r/[\p{L}a-zA-Z]+/)
  end

  defp question?(string) do
    string |> String.last == "?"
  end

  defp silence?(string) do
    string |> String.match?(~r/^\s*$/)
  end
end

File: clojure/bob/bob.clj
(ns bob
  (:require [clojure.string :as str]))

(defn shouting? [input]
  (and
    (= input (str/upper-case input))
    (not= input(str/lower-case input))))

(defn response-for [input]
  (cond
    (str/blank? input) "Fine. Be that way!"
    (shouting? input) "Woah, chill out!"
    (str/ends-with? input "?") "Sure."
    :else "Whatever."
  )
)
